### PR TITLE
[NVFuser] Upstream push 0811

### LINKED ISSUE #82784: NVFuser produces incorrect output for broadcasted dropout
```
import torch
import functorch
from functorch.compile import memory_efficient_fusion

def fn_drop(x, y):
    return torch.nn.functional.dropout(x) * y

x=torch.ones(4, device="cuda")
y=torch.ones(4,4,device="cuda")
opt_fn=memory_efficient_fusion(fn_drop)
for _ in range(100):
    out=opt_fn(x, y)
```

prints an output with randomly dropped out values instead of dropped out columns. Same for direct ts integration, don't know if it's the same for prim-based stack. 

cc @ezyang @gchanan @zou3519 @jjsjann123, @kevinstephano, @SherlockNoMad

### PR DESCRIPTION
**Pull request by jjsjann123**
Syncing nvfuser devel branch to upstream master. https://github.com/csarofeen/pytorch/

Code changes includes:

- codegen improvements:
  1. double support in expression evaluator
- bug fixes:
  1. dropout fix - rework RNG to support broadcasted dropout (Fixes #82784)
  2. expand fix - Patch expand+reduction, expand+view, rework view analysis and guard
- scheduler:
  1. manual transpose schedule example
  2. WIP transpose scheduler

Commits that's in this PR from the devel branch:

```
b7435afcd22c917713c2f41a7237bc26e1183f14 Transpose scheduler, step 1 (#1854)
8a45dbf72034684eb8e18b1835b533e90b68f184 Add an example on how to manually schedule transpose (#1889)
83dbf56a9554b2efbd5416461d938fff477b0b27 Patch dropout fix (#1898)
69d3519a532250719b1aa8341b50e067b181b42d Expand+Reduction, Expand+View support, rework View analysis and guards (#1883)
15091c488e96343bdc49e3990acbf238a3b3da51 Rework RNG to correctly support broadcasted dropout (#1888)
aafe2d048aaac596e503596a41303423619f3954 Make ExpressionEvaluator support Double (#1885)
```

RUN_TORCHBENCH: nvfuser

### CODE REVIEW

**Comment by jjsjann123** [comment]
Mostly wanted to push the broadcast/dropout fixes to patch #82784, But since we don't have many more PRs in devel yet. I'm just gonna push the whole thing. :crossed_fingers: Hopefully CI plays nicely for the small (~5k) PR here. :laughing:

**Comment by jjsjann123** [comment]
ah... forgot about TORCHBENCH... I'll try to push to upstream and restart this.

**Comment by jjsjann123** [comment]
closing in favor of #83239, so we can have TorchBench properly running.
